lib/ansible/modules/cloud/amazon/elb_target.py

```diff
@@ -185,19 +202,35 @@ def deregister_target(connection, module):
     except ClientError as e:
         module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
 
-    if target_description['TargetHealth']['Reason'] != "Target.NotRegistered":
-        try:
-            connection.deregister_targets(TargetGroupArn=target_group_arn, Targets=[target])
-            changed = True
-        except ClientError as e:
-            module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+    if 'State' in target_description['TargetHealth']:
+        if target_description['TargetHealth']['State'] not in ['draining', 'unused']:
+            try:
+                connection.deregister_targets(TargetGroupArn=target_group_arn, Targets=[target])
+                changed = True
+                if target_status:
+                    target_status_check(connection, module, target_group_arn, target, target_status, target_status_timeout)
+
+            except ClientError as e:
+                module.fail_json(msg="Ohh parps", exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
```

Comment:
Better message here than "Ohh parps"? ;)

Reply:
Set to use `e.message` 💯